monitoring/security/find-cves/cve_script.py

```diff
@@ -0,0 +1,158 @@
+    """
+    logging.info("Attempting to get jwt token from env")
+    jwt_token = os.getenv(env_variable)
+    if not jwt_token:
+        logging.error("Invalid JWT token provided")
+        raise ValueError("JWT token not found in environment variables")
+    return jwt_token
+
+
+def get_computer_ids(url: str, jwt_token: str) -> list[int]:
+    """
+    uses api of provided url to obtain computer ids of all computers in landscape
+
+    url       - domain of landscape server
+    jwt_token - jwt_token to use landscape api
+    """
+    logging.info("Attempting to get all computer ids")
+    url = url + "/api/v2/computers"
+
+    headers = {"Authorization": f"Bearer {jwt_token}"}
+
+    computer_ids = []
+    try:
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            computers = json.loads(json.dumps(response.json()))
+            computer_ids = [res["id"] for res in computers["results"]]
+        else:
+            logging.error(f"Received status code {response.status_code}")
+            logging.error(response.text)
+
+    except requests.exceptions.RequestException as e:
+        logging.error(f"An error occurred: {e}")
+
+    logging.info("Retrieved all computer ids")
+    return computer_ids
+
+
+def get_usns(url: str, jwt_token: str, ids: list[int]) -> dict:
+    """
+    retrieves all usns for every computer, along with package information
+    to identify which computer has the obtained usns
+
+    url       - domain of landscape server
+    jwt_token - jwt_token to use landscape api
+    ids       - list of all computer ids in landscape
+    """
+    logging.info("Attempting to obtain usns for all computers")
+    url = url + "/api/v2/usns?computer_ids="
+
+    for i in range(len(ids)):
+        url = url + f"{ids[i]}"
+        if i < len(ids) - 1:
+            url = url + ","
+
+    url = url + "&show_packages=true&limit=10000"
+
+    headers = {"Authorization": f"Bearer {jwt_token}"}
+
+    usn_results = []
+    try:
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            usns = json.loads(json.dumps(response.json()))
+            usn_results = usns["results"]
+        else:
+            logging.error(f"Received status code {response.status_code}")
+            logging.error(response.text)
+
+    except requests.exceptions.RequestException as e:
+        logging.error(f"An error occurred: {e}")
+
+    logging.info("Retrieved all usns")
+    return usn_results
+
+
+def get_computers_with_cve(usns: dict, cve: str) -> set:
+    """
+    finds which computers have the cve we are looking for within the usns
+
+    usns - usns for all computers in landscape
+    cve  - the cve we are looking for
+    """
+    computers = set()
+
+    for usn in usns:
+        cur_cves = usn["cves"]
+        for cur_cve in cur_cves:
+            if cur_cve["cve"] == cve:
+                parse_cur_packages(usn["packages"], computers)
```

Comment:
I'd get rid of this function, it's typically recommended against passing in an object to fill it, unless you have to. Here I would do something like:
for package in packages:
  computers.update(package["computer_ids"])